agent/src/control_interface/control_interface_task.rs

```diff
@@ -63,7 +66,31 @@ impl ControlInterfaceTask {
             authorizer,
         }
     }
+
+    // [impl->swdd~agent-closes-control-interface-on-missing-initial-hello~1]
+    async fn check_initial_hello(&mut self) -> Result<(), String> {
+        if let Ok(to_ankaios) = decode_to_server(self.input_stream.read_protobuf_data().await) {
+            match to_ankaios.try_into() {
+                Ok(ToAnkaios::Hello(to_ankaios::Hello { protocol_version })) => {
+                    check_version_compatibility(protocol_version)?
+                }
+                unexpected => {
+                    log::debug!("Expected initial Hello, received: '{unexpected:?}'.");
+                    return Err(INITIAL_HELLO_MISSING_MSG.into());
+                }
+            }
+        }
```

Comment:
One error case is ignored here and needs to be fixed.

Reply:
Changes are fine for me.